lib/string_basics.dart

```diff
@@ -200,6 +200,32 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a truncated version of the string.
```

Comment:
I renamed the parameters according to the suggestion. I made the `substitution` defaults to empty string so that by default it would appear that there is no `substitution` and so that the proposed `appendSubstitution` will not be needed anymore.

As for edge cases, I think the implementation right now covers them already minus the  `appendSubstitution`.

Reply:
Sorry, I think I explained kind of unclearly. Even without a default `substitution`, because there can be a user-provided `substitution`, we still end up with all the other edge cases, so we would still need a way to account for whether the user wants the character count length of their provided `substitution` to count towards `length` or not.

Reply:
I get it now and updated the logic. I also took a liberty on changing the `appendSubstitution` to `includeSubstitutionLength`, I fee like it does sound better for the intention of the parameter? But let me know if `appendSubstitution` do make sense better.